scripts/compare-statements.js

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+'use strict';
+
+/**
+ *
+ * Sort a list of compatibility statements based upon reverse-chronological order in the following order:
```

Comment:
It looks like this grouping is going to make the support blocks un-chronological in some cases. In particular I'm thinking of APIs that were in Presto or Edge, then prefixed-only in Chromium after the switch, and eventually unprefixed again, like this:

https://github.com/mdn/browser-compat-data/blob/3db2dcefb7d3e3cfd1fedb76116c74355c61501f/api/Element.json#L5870-L5882

I think the lint should at minimum allow this order, and ideally require it.

I have some guesses, but it would be good to spell out what the downsides would be of sorting all entries by version unconditionally.

Reply:
I think we can! First sort chronologically strictly, then find the "best" entry to show by default and move that to the beginnings.

Or we move this kind of logic to consumers, making this a much bigger project...

Reply:
I definitely think we should handle the logic here in BCD so that our consumers have less to worry about.

May I pass this PR off to you to implement the logic you recommended?

Reply:
Could we actually hold off on implementing this logic for another PR/migration?  While I think this would be great for us to do, I'm just not sure how to put it into code yet, and I'd love to get the order resolved so that our consumers aren't confused by the lack of order...